### ISSUE TITLE
Finished callback called on instantiation

### ISSUE THREAD

**Issue by Keats** [gilbarbara/react-joyride#103]
I start the tour in `componentDidMount` and witht the following jsx
```js
        <Joyride
          ref="joyride"
          steps={HOMEPAGE_TOUR}
          locale={locale}
          showBackButton={false}
          showSkipButton={true}
          callback={this.finishOnboarding.bind(this)}
          type="continuous" />
```
`finishOnboarding` is called with the following object upon instantiation:

```
Object {action: undefined, type: "finished", steps: Array[5], skipped: false}
```
I was using `completeCallback` originally and thought it was a bug with that but it seems that it isn't the case as it happens with `callback`

I can have a look at the code if you want, let me know first if this was somehow intended

**Comment by ericm546** [comment]
I have the same issue here.
It seem to be because toggleTooltip is called without an index in the start function
```
 if (autoStart) {
        this.toggleTooltip(true);
}

```
so in toggleTooltip:
`if (props.steps.length && !props.steps[index]) {`

become 
`if (props.steps.length && !props.steps[undefined]) {` 
and this return true

**Comment by ashkanjj** [comment]
Got the same issue here. It seems like the callback is fired only when you pass true to the start function like so: `this.refs.joyride.start()`.

**Comment by gilbarbara** [comment]
I think it's fixed in 1.4.8
Please reopen if not.

Thanks
